tests/unit/common/test_limiter.py

```diff
@@ -34,3 +34,31 @@ async def add_trace(start: str, sleep: float, stop: str):
         limiter.run(add_trace("s3", 0.03, "e3")),
     )
     assert trace == expected_trace
+
+
+@pytest.mark.parametrize(
+    "limit,expected_trace",
+    [
+        (1, ["s1", "e1", "s2", "e2", "s3", "e3"]),
+        (2, ["s1", "s2", "e2", "s3", "e3", "e1"]),
+        (3, ["s1", "s2", "s3", "e2", "e3", "e1"]),
+    ],
+)
+@pytest.mark.asyncio
+async def test_gather_limited(limit: int, expected_trace: Sequence[str]) -> None:
```

Comment:
interesting approach to testing the Limiter. it took a bit to understand, but I like it: no stubbing, no mocking, no exposure of internal state, only behaviour is tested.